src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Maestro.Web.Api.v2019_01_16.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.KeyVault;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    [Route("goals")]
+    [ApiVersion("2019-01-16")]
+    public class GoalController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        public GoalController(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        ///   Exposes methods to set goal time <see cref="Goal"/>
+        /// </summary>
+        /// <param name="ChannelName"></param>
+        /// <param name="DefinitionId"></param>
+        /// <param name="Minutes"></param>
+        [HttpPost]
+        [SwaggerApiResponse(System.Net.HttpStatusCode.Created, Type = typeof(Models.Goal), Description = "Goal for given Channel and Definition-Id is created/updated")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> Create([Required] int DefinitionId, [Required] string ChannelName, [Required] int Minutes)
+        {
+            Data.Models.Channel channel = await _context.Channels
+                    .Where(c => c.Name.Equals(ChannelName)).FirstOrDefaultAsync();
+            Data.Models.GoalTime goal = await _context.GoalTime
+                .Where(g => g.DefinitionId == DefinitionId && g.ChannelId == channel.Id).FirstOrDefaultAsync();
+
+            if (channel == null)
+            {
+                return NotFound();
+            }
+            if (goal == null)
+            {
+                var goalModel = new Data.Models.GoalTime
+                {
+                    DefinitionId = DefinitionId,
+                    Minutes = Minutes,
+                    ChannelId = channel.Id
+                };
+                await _context.GoalTime.AddAsync(goalModel);
+                await _context.SaveChangesAsync();
+                var goalTest = CreatedAtRoute(
+                    new
+                    {
+                        action = "CreateGoal",
+                        DefinitionId = DefinitionId,
+                        Minutes = Minutes,
+                        ChannelId = channel.Id
+                    },
+                    new Goal(goalModel));
+                return StatusCode((int)HttpStatusCode.Created);
```

Comment:
This should just always return Ok(goal), having it return different answers based on whether it exists or not isn't a useful distinction.

Reply:
I was returning 

1) 201 for created
2) 200 for updated

All the other APIs have create and update as different APIs, I thought it will give us a good distinction. May we can use this info later.

Reply:
I checked all the other methods, looks like we send a code 200 for all of them. So I will change this to send 200 for add as well as update. In future if we want, we can refactor to send the appropriate http code.

Reply:
It's not useful information, especially with the retry nature of the internet.  You might repost the "add", and get an update, which would be confusing. So reading it from the client is pointless (because you can't trust the value to be correct).

All web method should be idempotent, and return the same answer if the call them more than once (so that retrying) doesn't cause weird behavior.